.github/workflows/publish.yml

```diff
@@ -0,0 +1,32 @@
+name: Publish to NPM
+on:
+  release:
+    types: [published]
+npm:
+  name: Release NPM
+  runs-on: ubuntu-latest
+  needs: Version
+  environment: NPM Release
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: 'Check out the repo'
+        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+      - name: 'Install Node.js'
+        uses: actions/setup-node@e33196f7422957bea03ed53f6fbb155025ffc7b8 # v3.7.0
+        with:
+          node-version-file: '.nvmrc'
+          registry-url: 'https://registry.npmjs.org'
+          # Defaults to the user or organization that owns the workflow file
+          scope: '@waveshq'
+      - name: Install dependencies
+        run: npm ci
+      - name: Build the package
+        run: npm run build
+      - run: npm config set "//registry.npmjs.org/:_authToken" "\${NPM_AUTH_TOKEN}" --location=project
```

Comment:
```suggestion
      - run: npm config set "//registry.npmjs.org/:_authToken" "\${NODE_AUTH_TOKEN}" --location=project
```
?

Reply:
or we change line 30 to NPM_AUTH_TOKEN?

Reply:
This `NPM_AUTH_TOKEN` doesn't matter that much. It will look for secret.WAVESHQ_NPM_TOKEN.

Reply:
How can it look for secret.WAVESHQ_NPM_TOKEN?

Reply:
note that in the line 30, the env is NODE_AUTH_TOKEN, not NPM_AUTH_TOKEN

Reply:
I think better put NODE_AUTH_TOKEN here

Reply:
It doesn't make any difference. Line 30 is `WAVESHQ_NPM_TOKEN`. When `actions/setup-node` exec this will replace the given name here in .npmrc file [reference](https://docs.github.com/en/actions/publishing-packages/publishing-nodejs-packages). I have to push some other changes. So I will change this as well.